Upgrade curator library from 3.4 to 5.4

warrenvw opened the pull request:
Resolves #5.

The old curator library doesn't work with the latest versions of elasticsearch. This PR upgrades the curator library from version 3.4 to 5.4, and makes necessary changes to the pack. Some of the changes were non-trivial, and resulted in some backward incompatible changes (probably the most obvious of these are that some actions were removed).

Remaining Tasks:

 - [x] Implement and test remaining actions.
 - [x] Document changes.

warrenvw committed:
Initial commit for new curator lib

warrenvw committed:
Initial commit

warrenvw committed:
do_action() raises exception on failure

warrenvw committed:
Add new yaml files

warrenvw committed:
Static gen changes to lib/curator_actions.yaml

warrenvw committed:
Add new index actions

warrenvw committed:
In the middle of updating filters

warrenvw committed:
Add delete_aliases and missing command kwargs

warrenvw committed:
Bump version_min to 5.0.0

https://www.elastic.co/guide/en/elasticsearch/client/curator/current/version-compatibility.html

warrenvw committed:
Update invalid boolean values in YAML files

warrenvw committed:
Remove filter params from actions

warrenvw committed:
Updated lib/curator_actions.yaml

Then executed `python lib/static_metagen.py`, renamed files as
necessary.

warrenvw committed:
Remove execute bit on YAML files

warrenvw committed:
Remove lib/items_filter.py

warrenvw committed:
Add 'filters' to action YAML files

warrenvw committed:
Use curator to define actions and options

warrenvw committed:
Add command_on_cluster()

Kami commented:
It looks like that most of the non automatically generated code changes are in https://github.com/StackStorm-Exchange/stackstorm-elasticsearch/pull/9/files#diff-8a16f40b64b45d4ae45c64748f5c67be, right?

warrenvw commented:
Yes, that's correct.

warrenvw committed:
Added curator_json, rewrote index_closed

warrenvw committed:
Add curator_json to action YAML files

warrenvw committed:
Finish implementing search

warrenvw committed:
Remove execute bit on curator_actions.yaml

warrenvw committed:
iterate_filters calls filter_by_space

If the `space' filter type is specified, then filter_by_space() is called by
iterate_filters().

warrenvw committed:
Remove execute bit on a few files

warrenvw committed:
WIP solving "keywords must be strings" error

warrenvw committed:
Fix "keywords must be strings" error

Also, some other miscellaneous simplification and cleaning.

warrenvw committed:
Remove execute bit on .gitignore

warrenvw committed:
Add docs to methods, remove indices.delete.yaml

warrenvw committed:
Add process_action() and warn_if_no_indices

By using the curator method process_action(), we dramatically simplify
action handling within the pack.

warrenvw committed:
Remove debugging

warrenvw committed:
Add missing options

warrenvw committed:
Additional testing showed add/remove needed fix

Reindex action also needs some special handling

warrenvw committed:
Initial updates to README.md

warrenvw committed:
Initial version of basic system tests

warrenvw committed:
Change link to filter documentation
